### ISSUE TITLE
Haema compatibility

### ISSUE THREAD

**Issue by williambl** [florensie/artifacts-fabric#24]
**Is your feature request related to a problem? Please describe.**
Haema players have told me that they'd like the umbrella from artifacts to protect vampires from the sun, like how it protects origin phantoms.

**Describe the solution you'd like**
In Haema 1.6 I added [a set of events](https://github.com/williambl/haema/blob/master/src/main/kotlin/com/williambl/haema/api/VampireBurningEvents.kt) to accomplish this. The implementation would look something like:
```java
VampireBurningEvents.INSTANCE.getVETO().register(new VampireBurningEvents.Veto() {
    @NotNull
    @Override
    public TriState willVampireBurn(@NotNull PlayerEntity player, @NotNull World world) {
        return UmbrellaItem.isHeldUpInEitherHand(player) ? TriState.FALSE : TriState.DEFAULT;
    }

    @Override
    public int getPriority() {
        return 20;
    }
});
```

**Describe alternatives you've considered**
I *could* add it from my side, but there are other umbrella mods too and I wouldn't want to replicate the code for all mods.

**Comment by florensie** [comment]
Having the event already makes this way easier to support than Origins. Thanks for the suggestion, will implement it soon.

**Comment by florensie** [comment]
Fixed in 318beec7dda981d0f83cc9671e78cdab4aba27de

**Comment by williambl** [comment]
Awesome, thanks :)
